api/applications/applications_handler.go

```diff
@@ -582,59 +583,48 @@ func (ah *ApplicationHandler) getAdditionalRadixRegistrationUpdateValidators(cur
 
 // RegenerateDeployKey Regenerates deploy key and secret and returns the new key
 func (ah *ApplicationHandler) RegenerateDeployKey(ctx context.Context, appName string, regenerateDeployKeyAndSecretData applicationModels.RegenerateDeployKeyAndSecretData) error {
-	// Make check that this is an existing application and that the user has access to it
-	currentRegistration, err := ah.getUserAccount().RadixClient.RadixV1().RadixRegistrations().Get(ctx, appName, metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-
 	sharedKey := strings.TrimSpace(regenerateDeployKeyAndSecretData.SharedSecret)
-	updatedRegistration := currentRegistration.DeepCopy()
-	if len(sharedKey) != 0 {
-		updatedRegistration.Spec.SharedSecret = sharedKey
-	}
-
-	// Deleting SSH keys from RRs where these deprecated fields are populated
-	updatedRegistration.Spec.DeployKeyPublic = ""
-	updatedRegistration.Spec.DeployKey = ""
-
-	setConfigBranchToFallbackWhenEmpty(updatedRegistration)
-
-	err = ah.isValidRegistrationUpdate(updatedRegistration, currentRegistration)
-	if err != nil {
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Make check that this is an existing application and that the user has access to it
+		currentRegistration, err := ah.getUserAccount().RadixClient.RadixV1().RadixRegistrations().Get(ctx, appName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		updatedRegistration := currentRegistration.DeepCopy()
+		if len(sharedKey) != 0 {
+			updatedRegistration.Spec.SharedSecret = sharedKey
+		}
+		// Deleting SSH keys from RRs where these deprecated fields are populated
+		updatedRegistration.Spec.DeployKeyPublic = ""
+		updatedRegistration.Spec.DeployKey = ""
+		setConfigBranchToFallbackWhenEmpty(updatedRegistration)
+		if err := ah.isValidRegistrationUpdate(updatedRegistration, currentRegistration); err != nil {
+			return err
+		}
+		_, err = ah.getUserAccount().RadixClient.RadixV1().RadixRegistrations().Update(ctx, updatedRegistration, metav1.UpdateOptions{})
```

Comment:
Should we use reflect.DeepEqual to prevent unneccessary updates?

Reply:
Isn't it used? `updatedRegistration := currentRegistration.DeepCopy()`

Reply:
DeepEqual, not DeepCopy.

Example:

```
		if !reflect.DeepEqual(updatedRegistration, currentRegistration) {
			_, err = ah.getUserAccount().RadixClient.RadixV1().RadixRegistrations().Update(ctx, updatedRegistration, metav1.UpdateOptions{})
			return err
		}
		return nil
```